nav2_map_server/include/nav2_map_server/map_saver_server.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__MAP_SAVER_SERVER_HPP_
+#define NAV2_MAP_SERVER__MAP_SAVER_SERVER_HPP_
+
+#include "map_saver.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+#include "nav2_msgs/srv/save_map.hpp"
+
+namespace nav2_map_server
+{
+
+/**
+ * @class nav2_map_server::MapSaverServer
+ * @brief A class that provides map saving services. During
+ * the handling of service requests, MapSaver node is spinning
+ * inside to save the map.
+ */
+class MapSaverServer : public nav2_util::LifecycleNode
+{
+public:
+  /**
+   * @brief Constructor for the nav2_map_server::MapSaverServer
+   */
+  MapSaverServer();
+
+  /**
+   * @brief Destructor for the nav2_map_server::MapServerService
+   */
+  ~MapSaverServer();
+
+protected:
+  // Lifecycle interfaces
+  nav2_util::CallbackReturn on_configure(const rclcpp_lifecycle::State & state) override;
+  nav2_util::CallbackReturn on_activate(const rclcpp_lifecycle::State & state) override;
+  nav2_util::CallbackReturn on_deactivate(const rclcpp_lifecycle::State & state) override;
+  nav2_util::CallbackReturn on_cleanup(const rclcpp_lifecycle::State & state) override;
+  nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
+  nav2_util::CallbackReturn on_error(const rclcpp_lifecycle::State & state) override;
+
+  // A service to save the map to a file at run time (SaveMap)
+  rclcpp::Service<nav2_msgs::srv::SaveMap>::SharedPtr save_map_service_;
+
+  // The name of the service for saving a map from topic
+  static constexpr const char * save_map_service_name_{"save_map"};
```

Comment:
You end up casting this into a string in the source code, can we just make this a string? `static constexpr const` is alot of qualifiers for a 8 character string ;-)

Reply:
Fixed: https://github.com/ros-planning/navigation2/commit/84cff7d8cf43443e23bc4852129791d665f6e5c7#diff-e22d88faf6e8ac7a5338c8c6cbc2f7a4R80